hack/build-images.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used build new container images of the CAPV manager and
+# clusterctl. When invoked without arguments, the default behavior is to build
+# a new image for the manager and to tag it as `edge`
+#
+# Options:
+#   -p push the images to gcr.io
+#   -c tag the image as a CI image instead of the default
+#
+# Assumptions:
+#  If using -p to push images, then the env var GCR_KEY_FILE must be set
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# BASE_REPO is the root path of the image repository
+readonly BASE_REPO=gcr.io/cluster-api-provider-vsphere
+
+# CAPV_MANAGER_IMAGE is the path of the CAPV manager container image
+readonly CAPV_MANAGER_IMAGE=${BASE_REPO}/capv-release/manager
+
+# CAPV_MANAGER_IMAGE_CI is the path of the CAPV manager image for CI usage
+readonly CAPV_MANAGER_IMAGE_CI=${BASE_REPO}/capv-ci/manager
+
+# PUSH_IMAGE is a flag that, when set, denotes to push the images to the registry
+PUSH_IMAGE=
+
+# CI_IMAGE is a flag that, when set, denotes that the images are for CI purposes
+# only, and therefore are only tagged with the git hash and have a different
+# image name entirely
+CI_IMAGE=
+
+# IMAGE_NAME contains the name of the image that was built, so that it can be
+# pushed to the image registry
+IMAGE_NAME=
+
+function usage() {
+  cat <<EOF
+usage: ${0} [FLAGS]
+  Builds and optionally pushes new images for Cluster API Provider vSphere (CAPV)
+
+FLAGS
+  -c    the images are solely for the use by CI
+  -p    push the images to the public container registry
+EOF
+}
+
+
+function ensure_docker() {
+  if ! docker ps 1>/dev/null 2>&1 ; then
+    echo "Docker not available"
+    exit 1
+  fi
+}
+
+function build_image() {
+  IMAGE_NAME=${CAPV_MANAGER_IMAGE}:edge
+  local version
+  version=$(git describe --always --dirty --abbrev=8)
+  if [ "${CI_IMAGE}" ]; then
+    IMAGE_NAME=${CAPV_MANAGER_IMAGE_CI}:${version}
```

Comment:
So this produces image names like `gcr.io/cluster-api-provider-vsphere/capv-ci/manager:v0.3.0-60-g04abe3e1` for CI images, which is slightly different than what is happening today. Just want to point that change out.